odbc/src/api/data.rs

```diff
@@ -196,9 +198,58 @@ impl IntoCData for Bson {
     fn to_datetime(&self) -> Result<DateTime<Utc>> {
         match self {
             Bson::DateTime(d) => Ok((*d).into()),
-            Bson::String(s) => Utc
-                .datetime_from_str(s, "%FT%H:%M:%S%.3fZ")
-                .map_err(|_| ODBCError::InvalidDatetimeFormat(s.clone())),
+            Bson::String(s) => {
+                if s.contains('T') {
+                    // received a datetime string
+                    Utc.datetime_from_str(s, "%FT%H:%M:%S%.3fZ")
```

Comment:
Agreed, and which timestamp format(s) to support was something I was a bit unsure of. To clarify, do you think we should support `yyyy-mm-dd hh:mm:ss[.f...]` _instead_ of `yyyy-mm-ddThh:mm:ss[.f...]Z` or in addition to?

Reply:
I would support both because of the extended JSON representation of Bson Date looks like this : `{"$date": "2004-04-15T00:00:00Z"}`, so it can be handy if someone does a conversion to string and then back to datetime.
@pmeredit : WDYT?

Reply:
We definitely need to support `T` since that's what mongo uses. I'm not concerned for supporting the `T`-less version, but since you think it's good, I am fine with it :)

Should see if the parsing allows for optional characters.

Reply:
https://docs.rs/chrono/latest/chrono/format/strftime/index.html just noticed what we already have is covered completely with just `"%+"`. There are also convenience sequences to match the whole time, not just the date part. Should allow for cleaning up the code and supporting both formats pretty easily